aim/sdk/adapters/catboost.py

```diff
@@ -0,0 +1,94 @@
+from sys import stdout
+from typing import Optional
+
+from aim import Run
+from aim.ext.resource import DEFAULT_SYSTEM_TRACKING_INT
+
+
+class AimCallback:
+
+    def __init__(self, loss_function: Optional[str] = "Loss",
+                 repo: Optional[str] = None,
+                 experiment: Optional[str] = None,
+                 system_tracking_interval: Optional[int]
+                 = DEFAULT_SYSTEM_TRACKING_INT,
+                 log_system_params: Optional[bool] = True,
+                 log_cout=stdout):
+        super().__init__()
+        self._loss_function = loss_function
+        self._repo_path = repo
+        self._experiment = experiment
+        self._system_tracking_interval = system_tracking_interval
+        self._log_system_params = log_system_params
+        self._run = None
+        self._run_hash = None
+        self._log_cout = log_cout
+
+        if log_cout is not None:
+            assert hasattr(log_cout, "write")
+
+    @property
+    def experiment(self) -> Run:
+        if not self._run:
+            self.setup()
+        return self._run
+
+    def setup(self):
+        if self._run:
+            return
+        if self._run_hash:
```

Comment:
do we need the re-opening logic here? as I understand, the callback is used as a logger (much like the  stdout) so normally, it should be just deleted once the process is finished.

Reply:
leaving it wont have any negative impact I assume